### PR TITLE
feat: VectorMultiTrajectory memory statistics

### PR DESCRIPTION
**Pull request by paulgessinger**
See #1516

### CODE REVIEW

**Comment by paulgessinger** [comment]
This currently includes changes from #1507.

**Comment by andiwand** [comment]
are you planning to merge this @paulgessinger or is this just for debugging?

**Comment by paulgessinger** [comment]
I would like to merge this, but it is for debugging.

**Comment by paulgessinger** [comment]
@plariono this is the memory statistics feature that I mentioned. I think we can't easily backport this to v19 however, so not sure how easy it is for you to try it.

**Comment by paulgessinger** [comment]
Ok with added test coverage, codecov still believes the new functionality is largely only partially covered (by branch). I think I will override this here.

File: Examples/Algorithms/TrackFinding/src/TrackFindingAlgorithm.cpp [hunk 1]
```diff
@@ -104,13 +108,17 @@ ActsExamples::ProcessCode ActsExamples::TrackFindingAlgorithm::execute(
...
+  auto mtj = std::make_shared<Acts::VectorMultiTrajectory>();
+  auto results = (*m_cfg.findTracks)(initialParameters, options, mtj);
 
   // Compute shared hits from all the reconstructed tracks
   if (m_cfg.computeSharedHits) {
     computeSharedHits(sourceLinks, results);
   }
 
+  // Acts::VectorMultiTrajectory::Statistics::hist_t stats;
+
```
Comment (andiwand) [inline Examples/Algorithms/TrackFinding/src/TrackFindingAlgorithm.cpp:121]: do you want to keep that?

**Review by andiwand — APPROVED** [review]
lgtm

do we expect any production runtime overhead from these changes?

**Comment by paulgessinger** [comment]
I wouldn't think so since it only collects the stats once per event. Since it's trivial to turn off, I would think it's ok to merge.

File: Tests/UnitTests/Core/EventData/MultiTrajectoryTests.cpp [hunk 2]
```diff
@@ -1068,4 +1068,51 @@ BOOST_AUTO_TEST_CASE(MultiTrajectoryExtraColumnsRuntime) {
...
+  fillTrackState(pc, TrackStatePropMask::All, ts);
+
+  stats = mt.statistics();
+
+  for (int t = 0; t < type_axis.size(); t++) {
+    BOOST_TEST_CONTEXT((type_axis.bin(t) == 1 ? "meas" : "other"))
+    for (int c = 0; c < column_axis.size(); c++) {
+      std::string key = column_axis.bin(c);
+      BOOST_TEST_CONTEXT("column: " << key) {
+        // BOOST_CHECK_NE(h.at(c, t), 0);
```
[the remark this answers is not in the document]
Comment (paulgessinger) [reply Tests/UnitTests/Core/EventData/MultiTrajectoryTests.cpp:1107]:
```suggestion
```

**Comment by plariono** [comment]
> @plariono this is the memory statistics feature that I mentioned. I think we can't easily backport this to v19 however, so not sure how easy it is for you to try it.

Hi @paulgessinger, thank you. I've run 100 Pb-Pb events with this branch and it worked fine. I executed a full chain script similar to full_chain_itk.py. The crashes that I mentioned happened when running the ckf executable only with the Fatras csv output. 

Here is the output:

`
10:40:09    TrackFinding   INFO      TrackFindingAlgorithm statistics:
10:40:09    TrackFinding   INFO      - total seeds: 1009184
10:40:09    TrackFinding   INFO      - failed seeds: 15
10:40:09    TrackFinding   INFO      - failure ratio: 1.48635e-05
10:40:09    TrackFinding   INFO      Track State memory statistics (averaged):
other:
               count: 16481597.000000
               index:  1131.70M
             parPred:   754.47M
             covPred:  4526.81M
             parFilt:     0.00M
             covFilt:     0.00M
             parSmth:   754.47M
             covSmth:  4526.81M
                meas:   136.11M
             measCov:   816.66M
                 jac:  4526.81M
         sourceLinks:   148.43M
          projectors:    22.69M
               total: 17344.94M
meas:
               count: 11695718.00
               index:   803.08M
             parPred:   523.75M
             covPred:  3142.51M
             parFilt:   535.39M
             covFilt:  3212.32M
             parSmth:   535.39M
             covSmth:  3212.32M
                meas:   535.39M
             measCov:  3212.32M
                 jac:  3142.51M
         sourceLinks:   178.46M
          projectors:    89.23M
               total: 19122.69M
`

**Comment by paulgessinger** [comment]
@andiwand removed commented lines.

**Review by andiwand — APPROVED** [review]

**Pull request merged**